httpie/sessions.py

```diff
@@ -79,6 +81,12 @@ def update_headers(self, request_headers: RequestHeadersDict):
             if name == 'User-Agent' and value.startswith('HTTPie/'):
                 continue
 
+            if name.lower() == 'cookie':
+                for cookie_name, morsel in SimpleCookie(value).items():
+                    self['cookies'][cookie_name] = {'value': morsel.value}
```

Comment:
We added a new test, `test_cookie_storage_priority` in TestSessions to define and document the priority rules! Is this sufficient, or should there also be documentation in the docs?

Reply:
After our call this morning, @gmelodie and I discussed the possible behaviours of cookies expiring in sessions and which cookies should take precedent.

Our understanding was regardless of the expiry behaviour, the storage priority should still be server set > command line set > session file. For that reason, we didn't write further related tests. Here's a couple of example cases illustrating why we didn't think further tests were necessary:

-  Case 1: Server expires a cookie. If a server expires a cookie, it will be removed from the session file regardless of how it was set (`test_sessions.py` already has test covering this)
- Case 2: Cookies in session file expire. Cookies stored in a session file that expire will never be sent to the server and will be removed when the request is made
- Case 3: User sets an expired cookie in the command line. We haven't been able to find a way for a user to set an expired cookie in the command line. However, if one is set, we would assume the intention would be to remove the cookie from the session. As this cookie would replace the cookie in the session file, and expired cookies are cleared from the session file when the request is sent, this behaviour is satisfied.

Reply:
> Our understanding was regardless of the expiry behaviour, the storage priority should still be server set > command line set > session file. 

Agree 👍🏻